src/Asset/Asset.js

```diff
@@ -10,11 +11,7 @@ export class Asset {
   }
 
   async findByOwner(owner) {
-    return db.query(SQL`SELECT ${SQL.raw(this.tableName)}.*, (
-        ${PublicationQueries.findLastParcelPublicationJsonSql()}
-      ) as publication
-        FROM ${SQL.raw(this.tableName)}
-        WHERE owner = ${owner}`)
+    return db.query(await getFindByOwnerQuery(owner, this.tableName))
```

Comment:
I think we can get away without a `utils` here. The Estates query will need publications as well so maybe we can do something like this:

```javascript
async findByOwner(owner) {
    return await db.query(SQL`SELECT ${SQL.raw(this.tableName)}.*, (
        ${AssetQueries.findLastPublicationJsonSql(this.tableName)}
      ) as publication
        FROM ${SQL.raw(this.tableName)}
        WHERE owner = ${owner}`)

Reply:
I can rename `utils.js` to `AssetQueries` but my idea was to take the same approach as this one. According to the publications, yes, we're probably going to need them to estates to but there isn't a first `Auction` transaction as same as the estates right? Correct me if I'm wrong but doesn't the publications stands for a "parcel instance"?

Reply:
Already understood! what do you think on https://github.com/decentraland/marketplace/pull/224/commits/e43187928058505515865d9e692de7e890cf7ff2?